app/flow_io/decorators/spree_order_decorator.rb

```diff
@@ -5,36 +5,57 @@
 
 Spree::Order.class_eval do
 
-  # defines uniqe flow number per order
-  # format "solidus-order-hash" -> "s-o-hash"
+  # we now use Solidus number as Flow number, but I will this here for now
   def flow_number
     return self[:flow_number] unless self[:flow_number].blank?
     return unless id
+    number
+  end
 
-    token = ENV.fetch('SECRET_TOKEN')
-    number = 's-o-%s' % Digest::SHA1.hexdigest('%d-%s' % [id, token])
+  def flow_order
+    @_flow_hashie ||= Hashie::Mash.new(flow_cache['order'])
+  end
 
-    # fast update without callbacks
-    update_column :flow_number, number
+  # accepts line item, usually called from views
+  def flow_line_item_price line_item, total=false
+    id = line_item.variant.id.to_s
 
-    number
+    return 'Not synced with flow!' unless flow_cache['order']
```

Comment:
isn't it a bug if we expected a price and don't have it? means the item should not have been selected in the first place - in which case - yes, let's throw error